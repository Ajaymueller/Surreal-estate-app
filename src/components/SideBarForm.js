import React from 'react';
import '../styles/SideBar.css';

const SideBarForm = ({handleChange, handleSearch, query}) => {
    return (
        <div className="side-bar">
        <form onSubmit={handleSearch} data-testid="sideBarForm">
            <label htmlFor="search-input">
            <input type="text"
            value={query}
            onChange={handleChange} 
            data-testid="input-id"
            className="input"/>
            <button type="submit">Search</button>
            </label>
        </form>
        </div>
    )
}

export default SideBarForm;