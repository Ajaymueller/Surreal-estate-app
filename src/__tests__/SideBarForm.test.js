import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SideBarForm from '../components/SideBar';
import { MemoryRouter } from 'react-router-dom';
import "@testing-library/jest-dom/extend-expect";
import '@testing-library/jest-dom'

const props = {
    handleSearch: jest.fn(), 
    handleChange: jest.fn(), 
    query: "random", 
};

describe("SideBarForm", () => {
    xit("components renders with correct props", () => {
        const { asFragment } = render (
        <MemoryRouter><SideBarForm {...props}/></MemoryRouter> )
        expect(asFragment).toMatchSnapshot();
    });
    xit("renders a form", () => {
        const { getByTestId } = render (
        <MemoryRouter><SideBarForm {...props}/></MemoryRouter> )
        const form = getByTestId("sideBarForm");
        expect(form).toBeInTheDocument();
    });
});

describe("form", () => {
    xit("calls handleChange and changes the state on any change of input", () => {
        const { getByTestId } = render (
        <MemoryRouter><SideBarForm {...props}/></MemoryRouter> )
        const handleChange =  props.handleChange;
        const input = getByTestId("input-id"); 
        fireEvent.change(input, { target: { value: "random search" }});
        expect(input.value).toBe('random search');
        expect(handleChange).toHaveBeenCalled();
    });
    xit("calls handleSubmit on submission of form", () => {
        const { getByTestId } = render (
        <MemoryRouter><SideBarForm {...props}/></MemoryRouter> )
        const handleSearch = props.handleSearch;
        const form = getByTestId("sideBarForm");
        fireEvent.submit(form);
        expect(handleSearch).toHaveBeenCalled();
    })
});