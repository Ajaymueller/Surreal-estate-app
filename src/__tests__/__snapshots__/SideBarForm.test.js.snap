// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarForm components renders with correct props 1`] = `[Function]`;
